File: pdf-core/examples/generate_graphics.rs
use pdf_core::{Color, PdfDocument};

fn main() {
    let path = "graphics_output.pdf";
    let mut doc = PdfDocument::create(path).unwrap();
    doc.set_info("Creator", "rust-pdf");
    doc.set_info("Title", "Line Graphics Demo");
    doc.begin_page(612.0, 792.0);

    // Stroked rectangle (page border)
    doc.set_stroke_color(Color::rgb(0.0, 0.0, 0.0));
    doc.set_line_width(1.0);
    doc.rect(72.0, 72.0, 468.0, 648.0);
    doc.stroke();

    // Filled rectangle (light gray background box)
    doc.set_fill_color(Color::gray(0.9));
    doc.rect(100.0, 600.0, 200.0, 50.0);
    doc.fill();

    // Diagonal line
    doc.set_stroke_color(Color::rgb(0.0, 0.0, 1.0));
    doc.set_line_width(2.0);
    doc.move_to(100.0, 500.0);
    doc.line_to(300.0, 550.0);
    doc.stroke();

    // Triangle with fill and stroke
    doc.save_state();
    doc.set_fill_color(Color::rgb(1.0, 0.0, 0.0));
    doc.set_stroke_color(Color::rgb(0.0, 0.0, 0.0));
    doc.set_line_width(1.5);
    doc.move_to(350.0, 400.0)
        .line_to(450.0, 400.0)
        .line_to(400.0, 480.0)
        .close_path()
        .fill_stroke();
    doc.restore_state();

    // Nested rectangles using save/restore to isolate state
    doc.save_state();
    doc.set_stroke_color(Color::rgb(0.0, 0.5, 0.0));
    doc.set_line_width(3.0);
    doc.rect(150.0, 200.0, 300.0, 150.0);
    doc.stroke();

    doc.set_fill_color(Color::rgb(0.8, 0.9, 0.8));
    doc.rect(180.0, 230.0, 240.0, 90.0);
    doc.fill();
    doc.restore_state();

    // Add a label
    doc.place_text("Line Graphics Demo", 72.0, 740.0);

    doc.end_page().unwrap();
    doc.end_document().unwrap();
    println!("Generated: {}", path);
}

File: examples/php/generate_graphics.php
<?php
/**
 * Line graphics example.
 *
 * Mirrors: examples/rust/generate_graphics.rs
 *
 * Run with:
 *   php -d extension=target/release/libpdf_php.so examples/php/generate_graphics.php
 */

@mkdir(__DIR__ . '/../output', 0755, true);
$path = __DIR__ . '/../output/php-graphics.pdf';

$doc = PdfDocument::create($path);
$doc->setCompression(true);
$doc->setInfo("Creator", "rust-pdf-php");
$doc->setInfo("Title", "Line Graphics Demo");
$doc->beginPage(612.0, 792.0);

// Stroked rectangle (page border)
$doc->setStrokeColor(new Color(0.0, 0.0, 0.0));
$doc->setLineWidth(1.0);
$doc->rect(72.0, 72.0, 468.0, 648.0);
$doc->stroke();

// Filled rectangle (light gray background box)
$doc->setFillColor(Color::gray(0.9));
$doc->rect(100.0, 600.0, 200.0, 50.0);
$doc->fill();

// Diagonal line
$doc->setStrokeColor(new Color(0.0, 0.0, 1.0));
$doc->setLineWidth(2.0);
$doc->moveTo(100.0, 500.0);
$doc->lineTo(300.0, 550.0);
$doc->stroke();

// Triangle with fill and stroke
$doc->saveState();
$doc->setFillColor(new Color(1.0, 0.0, 0.0));
$doc->setStrokeColor(new Color(0.0, 0.0, 0.0));
$doc->setLineWidth(1.5);
$doc->moveTo(350.0, 400.0);
$doc->lineTo(450.0, 400.0);
$doc->lineTo(400.0, 480.0);
$doc->closePath();
$doc->fillStroke();
$doc->restoreState();

// Nested rectangles using save/restore to isolate state
$doc->saveState();
$doc->setStrokeColor(new Color(0.0, 0.5, 0.0));
$doc->setLineWidth(3.0);
$doc->rect(150.0, 200.0, 300.0, 150.0);
$doc->stroke();

$doc->setFillColor(new Color(0.8, 0.9, 0.8));
$doc->rect(180.0, 230.0, 240.0, 90.0);
$doc->fill();
$doc->restoreState();

// Add a label
$doc->placeText("Line Graphics Demo", 72.0, 740.0);

$doc->endPage();
$doc->endDocument();

echo "Generated: $path\n";
